File: cpp/Level/6/25206.cpp
#include <iostream>
#include <map>

using namespace std;

int main() {
    map<string, double> avg;
    avg.insert({"A+", 4.5});
    avg.insert({"A0", 4.0});
    avg.insert({"B+", 3.5});
    avg.insert({"B0", 3.0});
    avg.insert({"C+", 2.5});
    avg.insert({"C0", 2.0});
    avg.insert({"D+", 1.5});
    avg.insert({"D0", 1.0});
    avg.insert({"F", 0.0});

    string name, rate;
    double score;
    double all, result = 0;

    for (int i = 0; i < 20; i++) {
        cin >> name >> score >> rate;
        if (rate == "P") continue;
        all += score;
        result += score * avg[rate];;
    }

    cout << result / all;

    return 0;
}

File: kotlin/Level/6/25206.kt
fun main() {
    var avg = hashMapOf(
        "A+" to 4.5, "A0" to 4.0,
        "B+" to 3.5, "B0" to 3.0,
        "C+" to 2.5, "C0" to 2.0,
        "D+" to 1.5, "D0" to 1.0,
        "F" to 0.0
    )

    var all = 0.0
    var result = 0.0

    for (o in 1..20) {
        var x = readLine()!!.split(" ")
        if (x[2] == "P") continue
        all += x[1].toDouble()
        result += x[1].toDouble() * avg[x[2]]!!.toDouble()
    }

    print(result / all)
}